app/screens/manage_channel_members/manage_channel_members.tsx

```diff
@@ -0,0 +1,284 @@
+        },
+        loadingText: {
+            color: changeOpacity(theme.centerChannelColor, 0.6),
+        },
+        noResultContainer: {
+            flexGrow: 1,
+            flexDirection: 'row',
+            alignItems: 'center',
+            justifyContent: 'center',
+        },
+        noResultText: {
+            fontSize: 26,
+            color: changeOpacity(theme.centerChannelColor, 0.5),
+        },
+    };
+});
+
+const messages = defineMessages({
+    add_people: {
+        id: t('mobile.add_people.error'),
+        defaultMessage: "We couldn't add those users to the channel. Please check your connection and try again.",
+    },
+    button_manage: {
+        id: t('mobile.manage_members.manage'),
+        defaultMessage: 'Manage',
+    },
+    button_done: {
+        id: t('mobile.manage_members.done'),
+        defaultMessage: 'Done',
+    },
+});
+
+export default function ManageChannelMembers({
+    currentChannel,
+    componentId,
+    currentTeamId,
+    currentUser,
+    restrictDirectMessage,
+    teammateNameDisplay,
+    tutorialWatched,
+}: Props) {
+    const serverUrl = useServerUrl();
+    const theme = useTheme();
+    const style = getStyleFromTheme(theme);
+    const intl = useIntl();
+    const {formatMessage} = intl;
+
+    const currentUserId = currentUser.id;
+
+    const canManage = isSystemAdmin(currentUser.roles) || isChannelAdmin(currentUser.roles);
+    const searchTimeoutId = useRef<NodeJS.Timeout | null>(null);
+    const next = useRef(true);
+    const page = useRef(-1);
+    const mounted = useRef(false);
+
+    const [manageEnabled, setManageEnabled] = useState(false);
+    const [profiles, setProfiles] = useState<UserProfile[]>([]);
+    const [searchResults, setSearchResults] = useState<UserProfile[]>([]);
+    const [loading, setLoading] = useState(false);
+    const [term, setTerm] = useState('');
+
+    const isSearch = Boolean(term);
+
+    const loadedProfiles = ({users}: {users?: UserProfile[]}) => {
+        if (mounted.current) {
+            if (users && !users.length) {
+                next.current = false;
+            }
+
+            page.current += 1;
+            setLoading(false);
+            if (users?.length) {
+                setProfiles((prev: UserProfile[]) => [...prev, ...users]);
+            }
+        }
+    };
+
+    const clearSearch = useCallback(() => {
+        setTerm('');
+        setSearchResults([]);
+    }, []);
+
+    const getProfiles = useCallback(debounce(() => {
+        if (next.current && !loading && !term && mounted.current) {
+            setLoading(true);
+            fetchProfilesInChannel(serverUrl, currentChannel.id).then(loadedProfiles);
+        }
+    }, 100), [loading, isSearch, restrictDirectMessage, serverUrl, currentTeamId]);
+
+    const handleSelectProfile = useCallback(async (profile: UserProfile) => {
+        if (!manageEnabled) {
+            return;
+        }
+        const {user} = await storeProfile(serverUrl, profile);
```

Comment:
why this? as far as I'm seeing, `fetchProfilesInChannel` is already storing the profiles in the db, no ?

Reply:
Yes, you are correct.  

Can you recall why it was added in the user_list component?

https://github.com/mattermost/mattermost-mobile/blob/MM-47653-manage-members-screen/app/components/user_list/index.tsx#L167-L182

Reply:
Ahh..   Perhaps that is because it is used for create DM/GM when the profiles come from a search? In that case, all users on the server are available for adding to a GM, but you don't want to add them to wdb unless you create a GM with those selected profiles.

In the case of managing members in a channel, all the users are already in fetched and stored.  I removed the storeProfile call in the manage members use case you mentioned. Sound correct?

Reply:
Another question..  

currently, this component gets its channel members from pinging the server, but couldn't the user list just observe the cannel members from WDB? Shouldn't they be in the db already since you are currently viewing the channel?

The search is also querying getting user profiles through the searcprofiles API, but it could just filter the WDB user profiles based on the term, correct?